File: src/Game.js
'use strict';

function Game() {
  this.row_one =   ['-', '-', '-'];
  this.row_two =   ['-', '-', '-'];
  this.row_three = ['-', '-', '-'];
  this.turn = ['X', 'O'];
}

Game.prototype.makeMove = function (row, column) {
  var rowNum = this.rowSelection(row);
  rowNum[column -1] = this.turn[0];
};

Game.prototype.rowSelection = function (row) {
  if (row == 1) {
    return this.row_one;
  } else if (row == 2) {
    this.row_two;
  } else if (row == 3) {
    this.row_three;
  } else {
    throw new Error('This selection is invalid!');
  }
};

File: lib/game.rb
class Game

  attr_reader :row_one, :row_two, :row_three, :turn

  def initialize
   @row_one =   ['-', '-', '-']
   @row_two =   ['-', '-', '-']
   @row_three = ['-', '-', '-']
   @turn = [:X, :O]
  end

  def play(num, column)
   row_num = num_to_row(num)
   row_num[column -1] = @turn.first.to_s
   switch_player
  end


  private

  def switch_player
   @turn.reverse!
  end

  def num_to_row(num)
    hash = { 1 => @row_one,
             2 => @row_two,
             3 => @row_three}
    hash[num]
  end
end
